### ISSUE TITLE
Meta-request: mediawiki Math parity

### ISSUE THREAD

**Issue by Artoria2e5** [KaTeX/KaTeX#2876]
**Is your feature request related to a problem? Please describe.**
I am trying to [use KaTeX on Wikipedia for fun](https://phabricator.wikimedia.org/T99369#6970581). They have kicked away MathJax because of the server load and the speed issues, but I figured a KaTeX userscript won't hurt. After all, what's not to love about having actual text rendered instead of some SVG or even PNG image?

**Describe the solution you'd like:**
KaTeX should add support for everything TeX mentioned in the current [Wikipedia formula help page](https://en.wikipedia.org/wiki/Help:Displaying_a_formula). The errors are:
```
KaTeX parse error: Limit controls must follow a math operator at position 58: …ratorname {sn} \̲l̲i̲m̲i̲t̲s̲ ̲_{b>c}(b+c)
KaTeX parse error: Undefined control sequence: \S at position 63: …Re ,\circledS ,\̲S̲ ̲,\P ,\mathrm {\…
KaTeX parse error: Undefined control sequence: \P at position 9:  \amalg \̲P̲ ̲\S \%\dagger \d…
KaTeX parse error: Undefined control sequence: \sideset at position 2:  \̲s̲i̲d̲e̲s̲e̲t̲ ̲{_{1}^{2}}{_{3}…
KaTeX parse error: Undefined control sequence: \iiiint at position 2:  \̲i̲i̲i̲i̲n̲t̲ ̲\limits _{F}dx\…
KaTeX parse error: Undefined control sequence: \definecolor at position 2:  \̲d̲e̲f̲i̲n̲e̲c̲o̲l̲o̲r̲ ̲{myorange}{rgb}…
```

**Link to or name of a (La)TeX package that provides the same feature:**
* I have no idea what the \limits devil is.
* \S and \P are base LATEX 2ε, I think. They can be mapped to unicode chars.
* \sideset is from amsmath.
* \iiiint from esint is covered in #2059. 
* \definecolor from xcolor is covered in #750.

**Describe alternatives you've considered:**
Changing the existing formulae is unfortunately not acceptable.

**Additional context:**

**Comment by edemaine** [comment]
Neat request.  Is Wikipedia `<math>` running actual `latex` to produce SVG, or some other parser like MathJax? MathJax doesn't seem to support `\P` so I doubt it's that, but LaTeX should support `\operatorname*` so I'm confused.

* KaTeX actually supports `\operatorname*{sn}_{b>c}(b+c)`, but not the workaround.  I'm not sure how much it makes sense to support the workaround, but I can confirm that `\operatorname{sn}\limits_{b>c}(b+c)` does not crash in LaTeX, unlike KaTeX.  However, LaTeX doesn't seem to actually put the `b>c` below `sn`. On the other hand, `\mathop{\operatorname{sn}}\limits_{b>c}(b+c)` works fine in LaTeX and KaTeX.
* `\S` and `\P` are supported by KaTeX, but only in text mode (e.g. `\text{\S\P}`). It looks like they work in math mode in LaTeX, though, so we should fix that.
* `\sideset` seems to be a new request. [This issue](https://github.com/mathjax/MathJax/issues/1202) offers one approach to a definition. Because it's a TeX command, it's not easy to find a definition. 
* `\iiiint` is request #2059
* `\definecolor` is request #750

**Comment by Artoria2e5** [comment]
It's actually... a bit confusing about what they are using. They have this older actual-latex thing without Unicode support, but they have also touted a server-side MathJax deployment called Mathoid. Since Unicode is still broken, I guess they are still using latex.

**Comment by edemaine** [comment]
FWIW, server-side KaTeX rendering would probably be a lot lighter-weight than server-side actual LaTeX... 🙂

I'm confused why that page recommends a workaround for `\operatorname*`. Can you check whether `\operatorname*` actually works? If so, perhaps could remove the workaround...

**Comment by Artoria2e5** [comment]
Wow, it *is* broken! I guess they are using MathJax in the backend but still with an old validator or something (sigh).

**Comment by Artoria2e5** [comment]
WRT KaTeX on server-side, I doubt it will be applicable in mathoid's current form: mathoid uses MathJax's path SVG output, while KaTeX is HTML DOM-only (for good reason I feel). And that brings us to #2000 again.

(Honestly, I would rather avoid touching SVG with all the additional responsibilities in text layout. Especially SVG with paths instead of text.)

**Comment by edemaine** [comment]
Yeah, it would require switching away from SVG embeds, and move to inlined HTML + MathML.  I feel like the accessibility of screen reading MathML (which is presumably impossible for SVG) would be of interest to Wikipedia.

**Comment by Artoria2e5** [comment]
The definition for sideset looks fine on the playground. Maybe it's good to be added?

**Comment by Artoria2e5** [comment]
I just found out another cool thing about KaTeX: the HTML output allows line wrap between `.base` units!
